src/Mod/Draft/draftfunctions/scale.py

```diff
@@ -71,11 +71,9 @@ def scale(objectslist, scale=App.Vector(1,1,1),
         if hasattr(obj,'Shape'):
```

Comment:
I would replace it with check for `isinstance(obj, Part.Feature)`, where "Shape" is defined

Reply:
This is perhaps something for an additional PR?

Reply:
> For a more robust code, I would replace it with a check for `isinstance(obj, Part.Feature)`, where `Shape` is defined ...

This doesn't work perfectly with newer versions, as the introduction of `App::Link` (August 2019) changed the situation. `App::Links` are not `Part.Features` but still have a `Shape`, that is, they "link" to the shape of another object. Therefore, to support `App::Links` and working with assemblies, many tests for `Part::Feature` have been changed to `hasattr(obj, "Shape")`. If you have questions about this, discuss directly with realthunder.

Reply:
@vocx-fc Would it mean that `App::Link` acts as an instance of the class it points to? So it needs that `App::Link` to override `__instancecheck__` to return `True` for both the Link class and the class this link points to.

Or you're telling that `App::Link` has shape in addition to `Part.Feature`? In that case `App::Link` should just inherit from `Part.Feature`